src/S3/S3ClientTrait.php

```diff
@@ -213,15 +216,38 @@ public function determineBucketRegionAsync($bucketName)
         return $handler($command)
             ->then(static function (ResultInterface $result) {
                 return $result['@metadata']['headers']['x-amz-bucket-region'];
-            }, static function (AwsException $exception) {
-                $response = $exception->getResponse();
+            }, function (AwsException $e) {
+                $response = $e->getResponse();
                 if ($response === null) {
-                    throw $exception;
+                    throw $e;
+                }
+
+                if ($e->getAwsErrorCode() === 'AuthorizationHeaderMalformed') {
```

Comment:
Going to keep these separate since the Client handles the cache and the middleware generates into the coroutine instead of returning.